tests/components/device_tracker/test_owntracks.py

```diff
@@ -666,6 +671,34 @@ def test_events_only_off(self):
         # Location update processed
         self.assert_location_state('outer')
 
+    def test_event_gps_source_type_entry_exit(self):
+        """Test the entry event of gps source type."""
+        # Entering the owntrack circular region named "inner"
+        self.send_message(EVENT_TOPIC, REGION_GPS_ENTER_MESSAGE)
+
+        # Enter uses the zone's gps co-ords
+        self.assert_location_latitude(INNER_ZONE['latitude'])
+        self.assert_location_accuracy(INNER_ZONE['radius'])
+        self.assert_location_state('inner')
+
+        # source_type should be gps when enterings using gps.
+        self.assert_location_source_type('gps')
+
+
+        self.send_message(EVENT_TOPIC, REGION_GPS_LEAVE_MESSAGE)
+
+        # Exit switches back to GPS
+        self.assert_location_latitude(REGION_GPS_LEAVE_MESSAGE['lat'])
+        self.assert_location_accuracy(REGION_GPS_LEAVE_MESSAGE['acc'])
+        self.assert_location_state('outer')
+
+        # Left clean zone state
+        self.assertFalse(self.context.regions_entered[USER])
+
+        # source_type should be gps when leaving using gps.
+        self.assert_location_source_type('gps')
```

Comment:
Please write the tests so that the source type changes between messages. Otherwise it would be possible to remove the lines that send the second message without failing the test.